src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -820,14 +822,29 @@ private Task WriteSuffix()
                 _keepAlive = (connectionOptions & ConnectionOptions.KeepAlive) == ConnectionOptions.KeepAlive;
             }
 
+            // https://tools.ietf.org/html/rfc7230#section-3.3.1
+            // If any transfer coding other than
+            // chunked is applied to a response payload body, the sender MUST either
+            // apply chunked as the final transfer coding or terminate the message
+            // by closing the connection.
+            if (hasTransferEncoding && transferCoding != TransferCoding.Chunked)
+            {
+                _keepAlive = false;
+
+                if (hasConnection)
+                {
+                    responseHeaders.SetRawConnection("close", _bytesConnectionClose);
```

Comment:
I like setting _keepAlive to false, but I'm not sure about changing headers that the user as already explicitly set even to save them from violating the HTTP spec. Do we do that for other headers?